data/state/accountsDB.go

```diff
@@ -90,7 +95,7 @@ func (adb *AccountsDB) SaveAccount(account AccountHandler) error {
 
 	baseAcc, ok := account.(baseAccountHandler)
 	if ok {
-		err = adb.saveCode(baseAcc)
+		err = adb.saveCode(baseAcc, oldAccount)
```

Comment:
make a separate function to saveCode and saveDataTrie where you can handle all the cases.

Reply:
make type assertion for oldAccount as well. for save code you need baseAccountHandler only.

Reply:
Done.